app/assets/javascripts/govuk_publishing_components/lib/govspeak/youtube-link-enhancement.js

```diff
@@ -1,50 +1,106 @@
     var _this = this
+    var apiScriptInserted = false
+
+    for (var i = 0; i < $youtubeLinks.length; ++i) {
+      var $link = $youtubeLinks[i]
 
-    $youtubeLinks.each(function () {
-      var $link = $(this)
       // if users have disabled 'campaigns' cookie in the new cookie page settings
       // we also need to disable the youtube video embed
       if (_this.hasDisabledEmbed($link) || !YoutubeLinkEnhancement.campaignCookies()) {
         return true
       }
-
-      var videoId = YoutubeLinkEnhancement.parseVideoId($link.attr('href'))
+      // insert script only once, but only if a video in the list is set to embed
+      // and campaign cookies are enabled
+      if (!apiScriptInserted) {
+        insertApiScript(this.playList, this.players)
+        apiScriptInserted = true
+      }
+      var videoId = YoutubeLinkEnhancement.parseVideoId($link.getAttribute('href'))
       if (videoId) {
-        _this.embedVideo($link, videoId)
+        _this.setupVideo($link, videoId)
       }
-    })
+    }
+  }
+
+  var playerApiReady = function (playList, players) {
+    // youtube iframe api is very pick about knowing when it's ready,
+    // so they want this function to be on the window object
+    window.onYouTubePlayerAPIReady = function () {
+      // youtube API provides the variable when it is definitely ready
+      for (var i = 0; i < playList.length; i++) {
+        var currentPlayer = createVideoPlayer(playList[i])
+        players[i] = currentPlayer
+      }
+    }
+  }
+
+  var insertApiScript = function (playList, players) {
+    var tag = document.createElement('script')
+    tag.src = 'https://www.youtube.com/player_api'
+    var firstScriptTag = document.getElementsByTagName('script')[0]
+    firstScriptTag.parentNode.insertBefore(tag, firstScriptTag)
+
+    playerApiReady(playList, players)
   }
 
   YoutubeLinkEnhancement.prototype.hasDisabledEmbed = function ($link) {
-    return $link.attr('data-youtube-player') === 'off'
+    return $link.getAttribute('data-youtube-player') === 'off'
   }
 
-  YoutubeLinkEnhancement.prototype.embedVideo = function ($link, videoId) {
-    var $mediaPlayer = $('<span class="media-player" />')
+  YoutubeLinkEnhancement.prototype.setupVideo = function ($link, videoId) {
+    // Handle if the same video is inserted into the page more than once
+    var videoWrapperId = videoId + Math.random().toString()
 
-    $link.parent().replaceWith($mediaPlayer)
+    var parentPara = $link.parentNode
+    var parentContainer = parentPara.parentNode
 
-    var protocol = this.protocol()
+    var youtubeVideoContainer = document.createElement('div')
+    youtubeVideoContainer.className += 'youtube-video-container'
+    youtubeVideoContainer.innerHTML = '<span id="' + videoWrapperId + '"></span>'
+
+    parentContainer.replaceChild(youtubeVideoContainer, parentPara)
+    this.playList.push({
+      id: videoWrapperId,
+      videoId: videoId,
+      title: $link.textContent
```

Comment:
Could we add the word "video" in there somewhere? Maybe "Video of" in front of the link text? (Not sure if we need to ask a content designer about this.) One of the problems with the YouTube video player is that it doesn't announce anywhere that this is a video, you just find out by coming across a play button.
My initial proposal suggested to add a hidden heading before the iframe. Not sure why Jani didn't implement that. It would be okay to not have that if at least the title would say it's a video.

Reply:
@selfthinker We could add 'Video of:' to the link text. Would this be more confusing if, for example, the video title already included 'video'? For example, we have a video on GOV.UK with the title "Video guide for exporters seeking trade finance and insurance solutions" (https://www.gov.uk/government/collections/watch-our-video-guide)

Reply:
Have implemented the 'Video of: ' approach for now

Reply:
Is this the only text we insert as part of this process? I wonder if we want to consider that the content might not be English?

Although thinking about it that shipped might have already sailed if the whole YouTube interface is in English.

Reply:
@selfthinker not implemented as most videos have "Video of" in the title for the link-only fallback. If anything, those without should get updated.

@kevindew the API does provide a [hl parameter](https://developers.google.com/youtube/player_parameters#hl) with changes the player interface. 
I guess we would then need to either read the html lang or particular section lang attribute where the player is placed

Reply:
It would be good to know how many links/titles already have the word video in it and how likely that is. Regarding other languages, I guess adding " (video)" at the end is slightly safer as the word "video" is pretty much the same across most languages (rarely spelt differently). But the word "of" obviously isn't.